### ISSUE TITLE
Incorrect keyword argument mapping with **kwargs in aiohttp.ClientSession.get call

### ISSUE THREAD

**Issue by Miskler** [microsoft/pyright#10439]
## Environment data

* **Pylance version**: 2025.4.1
* **OS and version**: Linux Mint 21.3
* **Python version**: 3.10.12 64-bit (CPython), aiohttp 3.10.11

## Code Snippet

```python
import aiohttp

async def fetch(url: str, proxy: str | None = None) -> str:
    args = {"url": url}
    if proxy is not None:
        args["proxy"] = proxy

    async with aiohttp.ClientSession() as session:
        # ← Here Pylance reports:
        # "Argument of type 'str' cannot be assigned to parameter 'allow_redirects' of type 'bool'"
        async with session.get(**args) as response:
            return await response.text()
```

## Repro Steps

1. Open the above code in VS Code with the Pylance extension enabled.
2. Save the file and hover over the `session.get(**args)` invocation.
3. Observe the Pylance type-checking error under the `**args` expansion.

## Expected behavior

No type error should be reported. The `proxy` key in `**args` should map correctly to the `proxy: Optional[str]` parameter of `ClientSession.get()`.

## Actual behavior

Pylance attempts to assign the string value of `args["proxy"]` to the first keyword-only parameter (`allow_redirects: bool`) instead of `proxy`, and emits:

```
Argument of type "str" cannot be assigned to parameter "allow_redirects" of type "bool" in function "get"
```

**Comment by rchiodo** [comment]
I believe this is working as designed. The keyword 'proxy' argument isn't in the dictionary initially so Pylance can't tell that it should be a keyword argument. 

This makes Pylance find the argument:

```python
import aiohttp

async def fetch(url: str, proxy: str | None = None) -> str:
    args = {"url": url, "proxy": proxy }

    async with aiohttp.ClientSession() as session:
        # ← Here Pylance reports:
        # "Argument of type 'str' cannot be assigned to parameter 'allow_redirects' of type 'bool'"
        async with session.get(**args) as response:
            return await response.text()
```
